diff -urN driftnet-0.1.6-20040426cvs/Makefile driftnet-0.1.6-20040426cvs-patched/Makefile
--- driftnet-0.1.6-20040426cvs/Makefile	2007-10-22 12:51:34.000000000 +1100
+++ driftnet-0.1.6-20040426cvs-patched/Makefile	2012-11-03 00:33:42.584363996 +1100
@@ -35,7 +35,7 @@
 # Optional C compiler and linker flags. Typical driftnet builds have support
 # for displaying captured images in an X window, and need the following flags:
 CFLAGS  += `pkg-config --cflags gtk+-2.0`
-LDLIBS  += -ljpeg -lungif -lpng `pkg-config --libs gtk+-2.0`
+LDLIBS  += -ljpeg -lungif -lpng -lgif `pkg-config --libs gtk+-2.0`
 
 # Alternatively, you can build a version of driftnet which can only be used
 # in `adjunct' mode as the back end for some other image-processing program. To
